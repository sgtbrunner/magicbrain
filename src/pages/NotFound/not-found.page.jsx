import React from 'react';
import { useHistory } from 'react-router-dom';

import './not-found.styles.css';

const NotFound = () => {
  const history = useHistory();
  return (
    <div className="f3 fw6 pt6 pr6 pl6 black-80">
      <h1>Oooooppssss.......</h1>
      <h3>You were not supposed to see this!</h3>
      <h5>
        Click{' '}
        <button type="button" className="link-button" onClick={() => history.push('/')}>
          here
        </button>{' '}
        to go back
      </h5>
    </div>
  );
};
export default NotFound;
