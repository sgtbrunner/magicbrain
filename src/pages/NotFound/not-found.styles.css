.link-button {
  background: transparent;
  border: none;
  font-weight: bold;
  font-size: larger;
  color: white;
}
