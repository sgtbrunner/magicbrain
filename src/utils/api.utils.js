const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
