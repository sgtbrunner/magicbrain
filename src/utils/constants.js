export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
