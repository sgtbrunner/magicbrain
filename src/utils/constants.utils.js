const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
