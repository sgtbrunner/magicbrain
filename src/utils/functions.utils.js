export const getUserFromLocalStorage = () => localStorage.getItem('user');
