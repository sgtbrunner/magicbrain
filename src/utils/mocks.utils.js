export const USER = {
  id: 3,
  name: 'test',
  email: 'test@test.com',
  entries: 7,
};

export const OUTPUTS = {
  outputs: [
    {
      data: {
        regions: [
          {
            region_info: {
              bounding_box: {
                top_row: 0.5,
                left_col: 0.5,
                bottom_row: 0.5,
                right_col: 0.5,
              },
            },
          },
        ],
      },
    },
  ],
};

export const CREDENTIAL_ERROR = {
  error: 'You have entered an invalid username and/or password',
};

export const SERVER_ERROR = {
  error: 'Something went wrong. Please try again later.',
};
