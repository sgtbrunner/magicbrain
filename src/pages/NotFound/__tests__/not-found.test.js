import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import NotFound from '..';

const mountNotFoundPage = () => mountComponent({ component: NotFound });

const HEADING = 'heading';

describe('NotFound Page', () => {
  it('should match snapshot', () => {
    const { container } = mountNotFoundPage();

    expect(container).toMatchSnapshot();
  });

  it('should render default not found message on screen', () => {
    mountNotFoundPage();

    const message = screen.getByRole(HEADING, { name: /you were not supposed to see this!/i });
    expect(message).toBeInTheDocument();
  });

  it('should render three (3) headings', () => {
    mountNotFoundPage();

    const headings = screen.getAllByRole(HEADING);
    expect(headings).toHaveLength(3);
  });

  it('should render link button', () => {
    mountNotFoundPage();

    const linkButton = screen.getByRole('button', { name: 'here' });
    expect(linkButton).toBeInTheDocument();
  });
});
