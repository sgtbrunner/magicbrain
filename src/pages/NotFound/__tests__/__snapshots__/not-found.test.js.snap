// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page should match snapshot 1`] = `
<div>
  <div
    class="f3 fw6 pt6 pr6 pl6 black-80"
  >
    <h1>
      Oooooppssss.......
    </h1>
    <h3>
      You were not supposed to see this!
    </h3>
    <h5>
      Click
       
      <button
        class="link-button"
        type="button"
      >
        here
      </button>
       
      to go back
    </h5>
  </div>
</div>
`;
